ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -736,15 +743,18 @@ public void findEntriesSinceOnRestartTest() throws IOException, StoreException {
 
     UUID oldSessionId = state.sessionId;
     final MockId newId = state.getUniqueId();
+    short accountId = Utils.getRandomShort(TestUtils.RANDOM);
+    short containerId = Utils.getRandomShort(TestUtils.RANDOM);
     // add to allKeys() so that doFindEntriesSinceTest() works correctly.
     state.allKeys.put(newId, new Pair<IndexValue, IndexValue>(
-        new IndexValue(CuratedLogIndexState.PUT_RECORD_SIZE, firstRecordFileSpan.getStartOffset(), Utils.Infinite_Time),
-        null));
+        new IndexValue(CuratedLogIndexState.PUT_RECORD_SIZE, firstRecordFileSpan.getStartOffset(), Utils.Infinite_Time,
+            state.time.milliseconds(), accountId, containerId), null));
     state.recovery = new MessageStoreRecovery() {
       @Override
       public List<MessageInfo> recover(Read read, long startOffset, long endOffset, StoreKeyFactory factory)
           throws IOException {
-        return Collections.singletonList(new MessageInfo(newId, CuratedLogIndexState.PUT_RECORD_SIZE));
+        return Collections.singletonList(
+            new MessageInfo(newId, CuratedLogIndexState.PUT_RECORD_SIZE, accountId, containerId, Utils.Infinite_Time));
```

Comment:
Also how is this succeeding - i might have it wrong but the index value you added to `allKeys` has a op time different from the one you pass here. Have you updated all the comparison functions?